src/main/java/org/isf/telemetry/model/Telemetry.java

```diff
@@ -0,0 +1,162 @@
+ *
+ * https://www.gnu.org/licenses/gpl-3.0-standalone.html
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.isf.telemetry.model;
+
+import java.lang.reflect.Type;
+import java.time.LocalDateTime;
+import java.util.Map;
+
+import javax.persistence.Column;
+import javax.persistence.EmbeddedId;
+import javax.persistence.Entity;
+import javax.persistence.Table;
+import javax.persistence.Transient;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+
+/**
+ * @author Andrei
+ *
+ */
+@Entity
+@Table(name = "OH_TELEMETRY")
+public class Telemetry {
+
+	@EmbeddedId
+	private TelemetryId id;
+
+	// necessary for daemon. Used to understand if daemon is active. TRUE | FALSE | NULL
+	@Column(name = "TEL_ACTIVE")
+	private Boolean active;
+
+	@Column(name = "TEL_CONSENT")
+	private String consentData;
+
+	@Column(name = "TEL_INFO")
+	private String info;
+
+	@Column(name = "TEL_SENT_TIME")
+	private LocalDateTime sentTimestamp;
+
+	@Column(name = "TEL_OPTIN_DATE")
+	private LocalDateTime optinDate;
+
+	@Column(name = "TEL_OPTOUT_DATE")
+	private LocalDateTime optoutDate;
+
+	@Transient
+	private volatile int hashCode = 0;
+
+	public Telemetry() {
+	}
+
+	public LocalDateTime getSentTimestamp() {
+		return sentTimestamp;
+	}
+
+	public void setSentTimestamp(LocalDateTime sentTimestamp) {
+		this.sentTimestamp = sentTimestamp;
+	}
+
+	public Boolean getActive() {
+		return active;
+	}
+
+	public void setActive(Boolean active) {
+		this.active = active;
+	}
+
+	public LocalDateTime getOptinDate() {
+		return optinDate;
+	}
+
+	public void setOptinDate(LocalDateTime optinDate) {
+		this.optinDate = optinDate;
+	}
+
+	public LocalDateTime getOptoutDate() {
+		return optoutDate;
+	}
+
+	public void setOptoutDate(LocalDateTime optoutDate) {
+		this.optoutDate = optoutDate;
+	}
+
+	public String getConsentData() {
+		return consentData;
+	}
+
+	public void setConsentData(String consentData) {
+		this.consentData = consentData;
+	}
+
+	public String getInfo() {
+		return info;
+	}
+
+	public void setInfo(String info) {
+		this.info = info;
+	}
+
+	public int getHashCode() {
+		return hashCode;
+	}
+
+	public void setHashCode(int hashCode) {
+		this.hashCode = hashCode;
+	}
+
+	public Map<String, Boolean> getConsentMap() {
+		if (this.consentData != null && !this.consentData.isEmpty()) {
+			Type type = generateTypeToken().getType();
+			Gson gson = new Gson();
+			return gson.fromJson(this.getConsentData(), type);
+		}
+		return null;
+	}
+
+	private TypeToken<Map<String, Boolean>> generateTypeToken() {
+		return new TypeToken<Map<String, Boolean>>() {
+		};
```

Comment:
Why is this anonymous class empty?   Seem strange to me.

Reply:
I think old fashioned code, found that since Gson 2.8.0 one can use:

`Type type = TypeToken.getParameterized(Map.class, String.class, Boolean.class).getType();`

Thanks!